src/lib/pickles/step_verifier.ml

```diff
@@ -596,15 +596,20 @@ struct
              It should be sufficient to fork the sponge after squeezing beta_3 and then to absorb
              the combined inner product.
           *)
-          let num_commitments_without_degree_bound = Nat.N26.n in
+          let num_commitments_without_degree_bound = Nat.N41.n in
           let without_degree_bound =
             Vector.append
               (Vector.map sg_old ~f:(fun g -> [| g |]))
               ( [| x_hat |] :: [| ft_comm |] :: z_comm :: [| m.generic_comm |]
               :: [| m.psm_comm |]
               :: Vector.append w_comm
-                   (Vector.map sigma_comm_init ~f:(fun g -> [| g |]))
-                   (snd Plonk_types.(Columns.add Permuts_minus_1.n)) )
+                   (Vector.append (Vector.map m.coefficients_comm ~f:(fun g -> [| g |]))
+                      (Vector.map sigma_comm_init ~f:(fun g -> [| g |]))
+                      (snd Plonk_types.(Columns.add Permuts_minus_1.n)) )
+                   (snd
+                      Plonk_types.(Columns.add
+                         (fst (Columns.add Permuts_minus_1.n)) ) )
+              )
```

Comment:
is it worth splitting this in multiple `let ... in` statements? The indentation here makes it really hard to understand how this array is constructed

Reply:
I guess we're just appending everything:

```ocaml
let sg = Vector.map sg_old ~f:(fun g -> [| g |]) in
let coeffs_comm = Vector.map m.coefficients_comm ~f:(fun g -> [| g |]) in
let sigma = Vector.map sigma_comm_init ~f:(fun g -> [| g |]) in
let proof1 = snd Plonk_types.(Columns.add Permuts_minus_1.n) in
let coeff_sigma = Vector.append coeffs_comm sigma proof1 in
let proof2 = snd Plonk_types.(Columns.add (fst (Columns.add Permuts_minus_1.n))) in
let w_and_coeff_sigma = Vector.append w_comm coeff_sigma proof2 in
let rest = [| x_hat |] :: [| ft_comm |] :: z_comm :: [| m.generic_comm |] :: [| m.psm_comm |] :: w_and_coeff_sigma in
let proof3 = snd (Wrap_hack.Padded_length.add num_commitments_without_degree_bound) in
Vector.append sg rest proof3
```

I'm wondering why some can be appended with the normal `::` and some with `Vector.append`